### ISSUE TITLE
Give each piece of advice a permalink

### ISSUE THREAD

**Issue by bfirsh** [cli-guidelines/cli-guidelines#23]
Currently the main headings have permalinks, but it'd be neat if I could link directly to "Prefer flags to args." or whatever.

**Comment by shyndman** [comment]
I'd love to see this as well. This guide is easily adopted as a convention, and links would be helpful in code reviews/chats.

**Comment by shyndman** [comment]
Still thinking about this one.

Would it be acceptable to decorate the advice `<strong>`s with #s using JavaScript?

I'm not seeing a clear path to adding them in the generated markup. Goldmark has the ability to add attributes to some elements, but not strongs.

An alternate approach, where instead of strongs, h4s were used, would impact wrapping (although I guess we could float the paragraph that follows?)

Any thoughts?

**Comment by shyndman** [comment]
Styling is also a consideration. The existing headings are styled with a trailing # link. Would we want a similar treatment for the advice elements? 

My feeling is that having a # link per advice heading would add a lot of visual noise if they were always visible, but maybe not.

**Comment by tashian** [comment]
@shyndman I agree, a # link that appears only on hover would be great. I _love_ how the [Holloway Guides](https://www.holloway.com/g/equity-compensation) handle this situation — I think they have some JS that generates  per-paragraph anchors.
